lib/ReferencedResourceManager.js

```diff
@@ -81,6 +81,14 @@ module.exports = class ReferencedResourceManager {
 
       this._logger.info(`${this._data.type} event received ${this._selfLink} ${objectPath.get(this._data, 'object.metadata.resourceVersion')}`);
 
+      // Remove the resource from the FetchEnvs cache so that later FetchEnvs instances will get a fresh copy
```

Comment:
Does this._data have the contents of the resource? If so, we could update the cache here instead of faulting it in.

Reply:
Good point, trying it out...

Reply:
Updated PR to update the cache on added/updated/polled, and remove from cache on deleted (or unknown).